### DISCUSSION TITLE
Done status should reflect in close issue

### DISCUSSION THREAD

**Discussion by Nosfert** [community/community#7537] [Projects and Issues] [answered]
When you create issues they are added to a repo, and that great. 
When you change status on the issue in the board the status are only reflected as "Proejcts... status:Done" or whatever status you have done. 

Since a repo issue only have the status Open and Close I would suggest that we can pick if a status on the Project board should change the status on the Issue in the repo. 

That is changing status on the board would reflect on the Issue status. 

Since we can add "any number of statuses" 
I would suggest that each status have it's configurable status on the Issue in the repo. 

I might understand (have not tried this yet) why it's currently implemented the way it is. And I suspect that one could have the same issue in multiple projects, thus making it hard / impossible / bad if we can change the state of the Issue from multiple projects. 

But I do think that having the option would be good. 

At the moment we need to manually update the status on each issue on the Project and then go into each issue on the repo and change the status to Closed / Open. 

My suggestion as default setting, with the option of changing it however people want. 
<Todo> <Open>
<In Progress> <Open>
<Done> <Close>
<custom> <Choose one Open/Close>

**Comment by Nosfert** [comment]
@azenMatt  I know im being a bit sneaky pinging u here. 
But do you know if this is something that you have talked about?

**Reply by azenMatt** [reply]
Hi - this is something we are looking at for the next set of automation features. 

Right now the automation has this going one way (if you close and issue, it will get marked as done (or your preferred status) but I agree we need to also add in the automation to go the other way (so if an issue is marked as done, it gets closed)

**Reply by ghost** [reply]
This just came up in our internal conversation as well.
We wanted to mirror the current Project Kanban, so I've added "In Review" field but no PR/Issue automatically moves to DONE (or closed), which becomes a manual work that is a huge pain point.

It would be great if this can be added soon

**Reply by Nosfert** [reply]
@azenMatt any idea / timeline when this might be available feature? :) 
Biggest complaint so far in my team have been the lack of issue update when closing via the project view. 
At the moment I have to sit at the end of every sprint ant close issues that's been closed in the project view hehe, it's a "working solution" but not very scalable.

**Reply by azenMatt** [reply]
Sorry @Nosfert - no timelines on this yet

Not a great solution - but we are working on https://github.com/github/feedback/discussions/5381 at the moment which will hopefully at least make your manual work around less painful!

**Reply by chris-mcdonald-dev** [reply]
@azenMatt Just checking back a year later 🤞🏾, any chance this will get a timeline revisit?
Project boards have been great, but we're having the same tedious issue wrapping up every sprint.

**Reply by azenMatt** [reply]
Oh wow a year already 😅 

Let me check in with the team - I know this comes up a lot.

**Reply by azenMatt** [reply]
Ok - this is just about to be worked on so should be coming soon!

**Reply by H3nryM** [reply]
@azenMatt has this shipped?

**Reply by azenMatt** [reply]
No, it got caught up in problems related to: 

>I might understand (have not tried this yet) why it's currently implemented the way it is. And I suspect that one could have the same issue in multiple projects, thus making it hard / impossible / bad if we can change the state of the Issue from multiple projects.

and did not make it over the finish line. Let me ping the team again and see what the plan is to finish it.

**Reply by azenMatt** [reply]
I have also changed this to 'unanswered' which will help it's case with the team :) we use 'answered' to mean a fix has shipped. So often sort but most upvotes and unanswered to see what the top pain points are.

**Reply by H3nryM** [reply]
Thank you

**Reply by rantoniuk** [reply]
@azenMatt any updates?

**Comment by mohamed-modernise** [comment]
very well required feature

**Comment by JoseRamonRodriguez** [comment]
Really important feature!

**Comment by akurilin** [comment]
+1 On this one!

**Comment by joelveloso** [comment]
Anyone as an action for now?

**Reply by dasmerlon** [reply]
I've created an [action](https://github.com/dasmerlon/project-issue-state-sync) that closes or reopens issues if they have a specific project status.

Unfortunately, the workflow cannot be triggered when an issue is moved to a different project column, as there's no event one can subscribe to. But you can schedule it so that the issue states are updated regularly. :)

Note that this action only makes sense when you use the [built-in project workflows](https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/using-the-built-in-automations), which allow you to automatically move an issue to a specific project column when it's closed or reopened. Otherwise, when closing an issue, the old project status would be kept and the action would reopen the issue according to the project status.

**Comment by MaximeBelcour** [comment]
+1 on this one, we need it

**Comment by DannyDevSe** [comment]
Please add this if a issue is moved to done it should close the issue!

**Reply by caligo-erik** [reply]
I agree. 
I'm still trying out Github projects and managing my repos with Gitkraken.

I even reached out to Gitkraken with a bug report saying that all issues still show as Open, even if they are "Done" in the Github project. Only then I realized that moving the issue between columns in the project had no effect whatsoever on the issue, and you have to close every issue manually. This is bad.

**Comment by alper** [comment]
Are they doing any development on projects at all anymore?

**Comment by megaroeny** [comment]
Definitely need this, thank you!

**Comment by andyslack** [comment]
+1

**Comment by paresh-deshmukh** [comment]
+1 This is a very useful feature.

**Comment by przemyslaw** [comment]
+1 this would allow for more seamless project work.

**Comment by tmehlinger** [comment]
Happy new year, please implement this!

**Comment by morieeeenyo** [comment]
+1
very well required feature!!

**Comment by ezequielramos** [comment]
+1
please, implement this!

**Comment by nickponomar** [comment]
+1
Imho, an essential feature

**Comment by scarletquasar** [comment]
I approve the implementation, there is no reason to not implement it

**Comment by DaniJacq** [comment]
+1

**Comment by emil14** [comment]
+1

**Comment by wlmbj** [comment]
+1

**Comment by anders-jagard** [comment]
+1

**Comment by alisevych** [comment]
+1

**Comment by queenofcorgis** [comment, marked as answer]
Great news, we've shipped the new auto-close issues workflow in Projects! You can find more details here - https://github.blog/changelog/2024-04-25-github-issues-projects-auto-close-issue-project-workflow/

**Reply by Nosfert** [reply]
Looks great!! :)
